osfclient/models/storage.py

```diff
@@ -82,13 +83,18 @@ def create_file(self, path, fp, update=False):
             response = self._put(url, params={'name': fname}, data=fp)
 
         if response.status_code == 409:
-            if not update:
+            if not force and not update:
                 raise FileExistsError(path)
 
             else:
                 # find the upload URL for the file we are trying to update
                 for file_ in self.files:
                     if norm_remote_path(file_.path) == path:
+                        if not force:
+                            if sha256_checksum(path) == file_.hashes['sha256']:
```

Comment:
Does it happen that `file_.hashes` doesn't contain the key `sha256`? If it can happen we should use `file_.hashes.get('sha256')` everywhere.